machine-learning/text-count-vectorizer.ipynb

```diff
@@ -0,0 +1,212 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading Data\n",
+    "\n",
+    "As we'll see later, Dask-ML's `CountVectorizer` benefits from using the `dask.distributed` scheduler, even on a single machine."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask.distributed import Client\n",
+    "\n",
+    "client = Client(n_workers=4, threads_per_worker=1)\n",
+    "client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In this example, we'll work with the 20 newsgroups dataset from scikit-learn."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import sklearn.datasets\n",
+    "\n",
+    "news = sklearn.datasets.fetch_20newsgroups()\n",
+    "news['data'][:2]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This returns a list of documents (strings). Dask-ML's `CountVectorizer` expects a `dask.bag.Bag` of documents. We'll use `dask.delayed` to load the 20 newsgroups in parallel, taking care to load the data on the workers and not place large values (like `news['data']`) in the the task graph. See https://docs.dask.org/en/latest/best-practices.html#load-data-with-dask and https://docs.dask.org/en/latest/delayed-best-practices.html#don-t-call-dask-delayed-on-other-dask-collections for more on these concepts.\n",
+    "\n",
+    "This example is a bit contrived to get a Bag with multiple partitions. Typically the full dataset would be partitioned into multiple files on disk, and you'd load one partition per file. In this case, we split the single file into multiple partitions by loading the data and then slicing."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import dask\n",
+    "import numpy as np\n",
+    "import dask.bag as db\n",
+    "import toolz\n",
+    "\n",
+    "@dask.delayed\n",
+    "def load_news(slice_):\n",
+    "    \"\"\"Load a slice of the 20 newsgroups dataset.\"\"\"\n",
+    "    return sklearn.datasets.fetch_20newsgroups()['data'][slice_]\n",
+    "\n",
+    "npartitions = 10\n",
+    "partition_size = len(news['data']) // npartitions\n",
+    "\n",
+    "lengths = np.cumsum([partition_size] * npartitions)\n",
+    "lengths = [0] + list(lengths) + [None]\n",
+    "\n",
+    "slices = [slice(a, b) for a, b in\n",
+    "          toolz.sliding_window(2, lengths)]\n",
+    "# Notice the persist here! More details later.\n",
+    "documents = db.from_delayed([load_news(x) for x in slices]).persist()\n",
+    "documents"
```

Comment:
We could also call `db.read_sequence(..., npartitions=10).persist()` and then call `client.rebalance()`

Given that people are going to blindly copy-paste whatever we do anyway I'd personally rather that they see this.  It's a bit more in line with ordinary behavior I think.